compute_tools/src/lsn_lease.rs

```diff
@@ -0,0 +1,112 @@
+use anyhow::bail;
+use anyhow::Result;
+use postgres::{NoTls, SimpleQueryMessage};
+use std::{
+    str::FromStr,
+    sync::Arc,
+    thread,
+    time::{Duration, SystemTime},
+};
+
+use compute_api::spec::ComputeMode;
+use tracing::{error, info};
+use utils::lsn::Lsn;
+
+use crate::compute::{ComputeNode, ComputeState};
+
+pub fn launch_lsn_lease_loop_for_static(compute: &Arc<ComputeNode>) {
+    let lsn = {
+        let state = compute.state.lock().unwrap();
+        let spec = state.pspec.as_ref().expect("spec must be set");
+        match spec.spec.mode {
+            ComputeMode::Static(lsn) => lsn,
+            _ => return,
+        }
+    };
+    let compute = compute.clone();
+    thread::spawn(move || lsn_lease_loop(compute, lsn));
+}
+
+fn postgres_configs_from_state(compute_state: &ComputeState) -> Vec<postgres::Config> {
+    let spec = compute_state.pspec.as_ref().expect("spec must be set");
+    let conn_strings = spec.pageserver_connstr.split(',');
+
+    conn_strings
+        .map(|connstr| {
+            let mut config = postgres::Config::from_str(connstr).expect("invalid connstr");
+            if let Some(storage_auth_token) = &spec.storage_auth_token {
+                info!("Got storage auth token from spec file");
+                config.password(storage_auth_token.clone());
+            } else {
+                info!("Storage auth token not set");
+            }
+            config
+        })
+        .collect::<Vec<_>>()
+}
+
+fn lsn_lease_loop(compute: Arc<ComputeNode>, lsn: Lsn) {
+    loop {
+        let state = compute.state.lock().unwrap();
+
+        let spec = state.pspec.as_ref().expect("spec must be set");
+        let configs = postgres_configs_from_state(&state);
+
+        let cmd = format!("lease lsn {} {} {} ", spec.tenant_id, spec.timeline_id, lsn);
+
```

Comment:
```suggestion

drop(state);
```

Holding mutex across API call + `thread::sleep` might cause other threads to never acquire locks. 

Minimal example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6b11c0953905ac0d0e651cd5c188ef27

Reply:
Ah, thank you, yes. I limited scope of lock.